halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/ConfigCommand.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.cli.command.v1;
+
+import com.beust.jcommander.Parameters;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This is a top-level command for dealing with your halconfig.
+ *
+ * Usage is `$ hal config`
+ */
+@Parameters(commandDescription = "Configure and view your halconfig")
+public class ConfigCommand extends NestableCommand {
+  private Map<String, NestableCommand> subcommands;
+
+  public ConfigCommand(GlobalOptions globalOptions) {
+    super(globalOptions);
+
+    ShowCommand show = new ShowCommand(globalOptions);
+    this.subcommands = new HashMap<>();
```

Comment:
Is there ever a case where `subcommands` should be/is expected to be null instead of an empty list? If not, I'd suggest putting the `new HashMap` on the field declaration.